app/models/mapping_rule.rb

```diff
@@ -52,16 +52,25 @@ def mapping_blocks_as_json(input_version, output_version, mapping_rules)
         "output_taxonomy" => output_version,
         "rules" => mapping_rules.select do
           _1.input_version == input_version && _1.output_version == output_version
-        end.map(&:as_json),
+        end.map(&:as_json).compact_blank,
       }
     end
   end
 
   def as_json
-    resolve_input_attribute_values!
+    input_integration_version = input_version unless from_shopify?
+    output_integration_version = output_version if from_shopify?
+
+    input_json = input.as_json(integration_version: input_integration_version)
+    output_json = output.as_json(integration_version: output_integration_version)
+
+    if input_json.empty? || output_json.empty?
```

Comment:
This is a temporary solution to avoid writing mappings that do not have a corresponding category in shopify. There are currently some categories in the mappings that no longer exist in the shopify categories. In the future we will have checks for this scenario before writing to the data files, but as of now we have this scenario that we need to code against.